src/base/QXmppMessage.h

```diff
@@ -26,13 +26,18 @@
 #ifndef QXMPPMESSAGE_H
 #define QXMPPMESSAGE_H
 
-// Required for source compatibility
 #include "QXmppStanza.h"
 
+// Required for source compatibility
 #include <QDateTime>
 
+namespace std {
+template<typename T>
+class optional;
+}
```

Comment:
The macOS test is failing now, we should probably just include the optional instead of forward-declaring it here.